### ISSUE TITLE
Windows 10. Cannot use jpackage with GraalVM 21.

### ISSUE THREAD

**Issue by scientificware** [oracle/graal#3948]
**Describe the issue**
- When using openjdk-17, jpackage works as expected.
- But when we use graalvm-ce-java17-21.3.0 or graalvm-ce-java16-21.1.0 there is an exception.

**Steps to reproduce the issue**
1. The code :
```
package hello;

import java.awt.Frame;
import java.awt.Label;
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;

public class HelloAWT {

    public static void main(String[] args) {
        System.out.println("java.library.path : \"" + System.getProperty("java.library.path") + "\"");
        Frame f = new Frame( "Hello world!" );
        Label label = new Label("Hello World", Label.CENTER);
        f.add(label);
        f.addWindowListener(
            new WindowAdapter(){
                public void windowClosing( WindowEvent e ){
                    System.exit( 0 );
                }
            });
        f.setSize( 300, 100 );
        f.setVisible(true);
        System.out.println("Hello World!!");
    }
}
```
2. Compile :
>`javac -d . HelloAWT.java`

3. Create jar file :
>`jar --create --verbose --file HelloAWT.jar --main-class hello.HelloAWT hello`

4. Create exe file :
>`jpackage --input . --name helloawt --app-version 1.0 --win-dir-chooser --win-menu --win-console --main-jar HelloAWT.jar`

**GraalVM and my environment:**
 - GraalVM version : 21.1 or 21.3
 - JDK major version : 16 or 17
 - OS: **Windows 10**
 - Architecture: **AMD 64**

**Output**

```
java.lang.module.FindException: Hash of jdk.jpackage (7b29cf91a3423c4473757d30e37d6d6f8cf568a70f6f111682431396df7ab38f) differs to expected hash (c3b4618bbb094fe2cb009cc29e0f0e182c8fe89ff3001bf7cbe4f0612c394706) recorded in java.base
```

**Comment by unlucku** [comment]
Having the same issue here, cannot take leverage of graal vm's runtime on my applications because of this :/

**Comment by unlucku** [comment]
Found a temporary workaround!
This only happens when jlink tries to add jdk.jpackage to --add-modules argument.
If your application is not using the jdk.jpackage module (Yes you are using it for building, but are you using it IN YOUR APP, like the code you specified does not use the jdk.jpackage imports, which means you are not using it in the app)

Therefore you can exclude it by using the following argument to your jpackage call:
```jpackage ... --add-modules jdk.management.jfr,java.rmi,jdk.jdi,jdk.charsets,jdk.xml.dom,java.xml,java.datatransfer,jdk.jstatd,jdk.httpserver,java.desktop,org.graalvm.locator,java.security.sasl,jdk.zipfs,java.base,jdk.crypto.ec,jdk.javadoc,org.graalvm.truffle,jdk.management.agent,jdk.jshell,com.oracle.graal.graal_enterprise,jdk.editpad,jdk.sctp,java.sql.rowset,jdk.jsobject,jdk.unsupported,java.smartcardio,jdk.jlink,org.graalvm.sdk,java.security.jgss,jdk.nio.mapmode,java.compiler,jdk.dynalink,jdk.unsupported.desktop,jdk.accessibility,jdk.security.jgss,jdk.incubator.vector,java.sql,java.transaction.xa,java.xml.crypto,java.logging,jdk.jfr,jdk.internal.vm.ci,jdk.crypto.cryptoki,jdk.net,jdk.random,java.naming,jdk.internal.vm.compiler.truffle.jfr,jdk.internal.ed,java.prefs,java.net.http,jdk.compiler,jdk.internal.opt,jdk.naming.rmi,jdk.jconsole,jdk.attach,org.graalvm.js.scriptengine,jdk.crypto.mscapi,jdk.internal.le,java.management,jdk.jdwp.agent,jdk.internal.jvmstat,jdk.incubator.foreign,java.instrument,jdk.internal.vm.compiler.management,jdk.internal.vm.compiler,jdk.management,jdk.security.auth,java.scripting,jdk.jdeps,jdk.jartool,java.management.rmi,jdk.naming.dns,jdk.localedata```

These are all the jmods for the 21.3.0 oracle java 17 graal build. To check which ones you need, you have two options:

1. Either make it so you do not use the --main-jar command and use its alternative --module (I have not tested this as my app uses --main-jar)
2. If you do use --main-jar, then simply pass the --add-modules command like I explained above (With your versions jmods while exclude jpackage jmod). This works well because if you dont add the --add-modules argument, by default it uses all the jmods. So by using just specifiying all of them but the jpackage jmod, you can get around this issue! For me it did take a bit longer to build for some reason, and it also did not have any output till later, so keep that in mind.

NOTE: IF YOUR APP ACTUALLY USES JDK.JPACKAGE AT RUNTIME, THESE FIXES WILL NOT WORK FOR YOU!!!

**Comment by scientificware** [comment]
Hi @unlucku, great thanks for your suggestion. :smiley:

Workaround :

4. Check the modules with `jdeps HelloAWT.jar`

```
HelloAWT.jar -> java.base
HelloAWT.jar -> java.desktop
   hello     -> java.awt                                           java.desktop
   hello     -> java.awt.event                                     java.desktop
   hello     -> java.io                                            java.base
   hello     -> java.lang                                          java.base
   hello     -> java.lang.invoke                                   java.base
```

5. Use it to build a custom runtine with `jlink` or package with the `--add-modules` option in `jpackage` :

> `jpackage --add-modules java.awt,java.desktop --input . --name helloawt --app-version 1.0 --win-dir-chooser --win-menu --win-console --main-jar HelloAWT.jar`

**Comment by ansalond** [comment]
This should be fixed in the upcoming January releases (22.0.0, 21.3.1).

**Comment by scientificware** [comment]
@ansalond  Thanks. 😊

**Comment by scientificware** [comment]
Works fine with 22.0.0.2.

**Comment by scientificware** [comment]
Fixed, works fine with 22.0.0.2.
